dev/Security.AccessControl/Security.AccessControl/GetSecurityDescriptorForAppContainerNames.cpp

```diff
@@ -0,0 +1,116 @@
+﻿#include "pch.h"
+
+#include <wil/resource.h>
+#include <wil/result.h>
+#include "Security.AccessControl.h"
+#include <vector>
+#include <sddl.h>
+#include <AclAPI.h>
+#include <UserEnv.h>
+
+using namespace std;
+
+namespace
+{
+    wil::unique_any_psid GetCurrentProcessSid()
+    {
+        wil::unique_any_psid result;
+        wil::unique_process_heap_ptr<TOKEN_GROUPS> tg;
+        DWORD length = 0;
+        if (!::GetTokenInformation(
+            ::GetCurrentProcessToken(),
+            TokenLogonSid,
+            nullptr,
+            0,
+            &length
+        ))
+        {
+            auto err = ::GetLastError();
+            if (err != ERROR_INSUFFICIENT_BUFFER)
+            {
+                THROW_WIN32(err);
+            }
+            tg.reset(static_cast<TOKEN_GROUPS*>(::HeapAlloc(::GetProcessHeap(), HEAP_ZERO_MEMORY, length)));
+        }
+
+        THROW_IF_WIN32_BOOL_FALSE(::GetTokenInformation(
+            ::GetCurrentProcessToken(),
+            TokenLogonSid,
+            tg.get(),
+            length,
+            &length));
+
+        auto sidLength = ::GetLengthSid(tg->Groups[0].Sid);
+        result.reset(::HeapAlloc(::GetProcessHeap(), 0, sidLength));
+        THROW_IF_WIN32_BOOL_FALSE(::CopySid(
+            sidLength, result.get(), tg->Groups[0].Sid));
+        return result;
+    }
+}
+
+STDAPI GetSecurityDescriptorForAppContainerNames(
+    uint32_t accessRequestCount,
+    _In_reads_(accessRequestCount)
+        const AppContainerNameAndAccess* accessRequests,
+    _In_opt_ PSID principal,
+    uint32_t principalAccessMask,
+    _Outptr_ PSECURITY_DESCRIPTOR* securityDescriptor
+)
+{
+    *securityDescriptor = nullptr;
+    try
+    {
+        vector<EXPLICIT_ACCESS> ea;
+        if (accessRequestCount == UINT32_MAX)
```

Comment:
Consider removing this "overflow' check; if the reserve fails then eventually the push-backs will fail later and that's OK.